llvm/lib/Target/DirectX/DXILResourceAccess.cpp

```diff
@@ -198,6 +203,111 @@ static void createLoadIntrinsic(IntrinsicInst *II, LoadInst *LI, Value *Offset,
   llvm_unreachable("Unhandled case in switch");
 }
 
+static void collectBlockUseDef(Instruction *Start,
+                               SmallVectorImpl<Instruction *> &Out) {
```

Comment:
Should this just return a `SmallVector<Instruction *>` rather than fill `Out`? I feel like that would be slightly clearer.